src/modules/new-request-form/fields/MultiSelect.tsx

```diff
@@ -0,0 +1,174 @@
+  const [activeOptions, setActiveOptions] = useState<INestedOption[] | undefined>(nestedOptions.root);
+
+  const handleChange: IComboboxProps['onChange'] = changes => {
+    if(changes.selectionValue !== undefined) {
+      // for an option like `Color::Special::Radioactive Green` the return will be `Radioactive Green`
+      const lastSelectedItem : string = (changes.selectionValue as OptionValue[]).slice(-1).toString();
+      if (lastSelectedItem == 'back') {
+        // walks back the subgroup option chain. Example: from: `Color::Special` to ``Color`
+        subGroupStack.pop();
+        const previousSubGroup: string | undefined = subGroupStack.length > 0 ? subGroupStack.slice(-1)[0]  : "root";
+        setSubGroupStack(subGroupStack);
+        previousSubGroup == "root" ? setActiveSubOption(null) : setActiveSubOption(previousSubGroup!);
+        setActiveOptions(nestedOptions[previousSubGroup!]);
+      } else if (nestedOptions[lastSelectedItem] !== undefined) {
+        // the selected Item represents/matches an option subgroup then move up the group chain
+        // Example: if lastSelectedItem = `Color`, the component move up the group chain from root to color
+        // and it will update the activeOptions property to display the elements inside `nestedOptions['Color']
+        if (!subGroupStack.includes(lastSelectedItem)) {
+          subGroupStack.push(lastSelectedItem)
+          setSubGroupStack(subGroupStack);
+          setActiveSubOption(lastSelectedItem);
+          setActiveOptions(nestedOptions[lastSelectedItem])
+        }
+      } else {
+        // if the lastSelectedItem represents a option Value then we update the component state
+        setSelectValues((changes.selectionValue as string[]))
+      }
+    }
+  };
+
+  return (
+    <GardenField>
+      {
+        (selectedValues as Array<string>).map((selectedValue) => (
+          <input type="hidden" name={`${name}[]`} value={selectedValue} />
+        ))
+      }
+      <Label>{label}</Label>
+      {description && <Hint>{description}</Hint>}
+      <Combobox
+        isMultiselectable
+        inputProps={{ required }}
+        isEditable={false}
+        validation={error ? "error" : undefined}
+        onChange={handleChange}
+        selectionValue={selectedValues}
+      >
+        {activeSubGroup && <OptGroup key={activeSubGroup} label={activeSubGroup} value="back"></OptGroup>}
+        {activeSubGroup && <Option key="back" value="back" label={activeSubGroup} type="previous">Back</Option>}
+        {activeOptions?.map((option) => (
+          <Option
+            key={option.value}
+            value={option.value!}
+            label={option.label}
+            type={option.type}
+            isSelected={(selectedValues as Array<string>).includes(option.value!.toString())}
+          >
+            {option.name}
+          </Option>
+        ))}
+
+        {activeSubGroup && <OptGroup/>}
```

Comment:
The rendered markup doesn't align with the [nested example in the Garden website](https://garden.zendesk.com/components/combobox#nested).  

I think that we should render something like this when we are in a nested level, to be aligned with the example and probably have better A11Y:
```tsx
<Option value="back" type="previous">Back</Option>
<OptGroup aria-label="Color">
   <Option ...></Option>
   <Option ...></Option>
</OptGroup>
```

while right now we are rendering:
```tsx
<OptGroup label="Color" value="back"></OptGroup>
<Option value="back" label="Color" type="previous">Back</Option>
<Option ...></Option>
<Option ...></Option>
<OptGroup />
```